docs/user/kafka/consumers.rst

```diff
@@ -0,0 +1,159 @@
+``KafkaDataStore``. As ``SimpleFeatures`` are created, modified,
+deleted, or cleared by the Kafka Data Store Producer, the current state
+is updated. All queries to the ``SimpleFeatureSource`` are queries
+against the current state. For example:
+
+.. code-block:: java
+
+    String typeName = ...
+    SimpleFeatureSource liveFeatureSource = consumerDs.getFeatureSource(typeName);
+
+    Filter filter = ...
+    liveFeatureSource.getFeatures(filter);
+
+It is also possible to provide a CQL filter to the getFeatureSource method call which will ensure
+the resulting ``FeatureSource`` only contains certain records. Providing a filter to reduce the number of
+returned records will provide a performance boost when using the featureSource.
+
+.. code-block:: java
+
+    String typeName = ...
+    SimpleFeatureSource liveFeatureSource = consumerDs.getFeatureSource(typeName, filter);
+
+Replay Mode
+-----------
+
+Replay mode allows the a user to query the ``KafkaDataStore`` as it
+existed at any point in the past. Queries against a Kafka Replay Simple
+Feature source specify a historical time to query and only the set and
+version of ``SimpleFeature``\ s that existed at that point in time will
+be used to answer the query.
+
+In order to use Replay mode some additional hints are required: the
+start and end times of the replay window and a read behind duration:
+
+.. code-block:: java
+
+    Instant replayStart = ...
+    Instant replayEnd = ...
+    Duration readBehind = ...
+    ReplayConfig replayConfig = new ReplayConfig(replayStart, replayEnd, readBehind);
+
+The replay window is simply an optimization that allows the Kafka Replay
+Feature Source to load, at initialization time, all state changes that
+occur within the window. Any query for a time outside of the window will
+return no results even if features existed at that time.
+
+The read behind is the amount of time used to rebuild state. For
+example, if ``readBehind = 5s`` then for a query requesting state at
+``time = t`` all state changes that occurred between ``t - 5s`` and
+``t`` will be used to build the state at time ``t`` which will then be
+used to answer the query. Selecting an appropriate read behind requires
+an understanding of the producer. The expected uses case is a producer
+that updates every simple feature, even if it hasn't changed, at a
+regular interval. For example, if the producer is updating every ``x``
+seconds then a read behind of ``x + 1s`` might be appropriate.
+
+During initialization of the Kafka Replay Feature Source all state
+changes from ``replayStart - readBehind`` to ``replayEnd`` will be read
+and cached. As the size of the replay window and read behind increases
+so does the amount of data that must be read and cashed. So, both the
+size of the window and the read behind should be kept as small as
+possible.
+
+After creating the ``ReplayConfig`` pass it, along with the
+``streamingSFT`` to the ``KafkaDataStoreHelper``:
```

Comment:
Awkward comma, maybe?: 
After creating the ``ReplayConfig``, pass it, along with the ``streamingSFT``, to the ``KafkaDataStoreHelper``: